frontend/localforageSetup.js

```diff
@@ -0,0 +1,12 @@
+(function () {
```

Comment:
I think it would be better to do
`(function startIDB() {
         ...
  })();`